src/PowerShellEditorServices/Services/DebugAdapter/DebugService.cs

```diff
@@ -263,7 +264,30 @@ public VariableDetailsBase[] GetVariables(int variableReferenceId)
                 VariableDetailsBase parentVariable = variables[variableReferenceId];
                 if (parentVariable.IsExpandable)
                 {
-                    childVariables = parentVariable.GetChildren(_logger);
+                    // PERFORMANCE: Because of ToString calls that might be PS script properties, we want this
+                    // to run on the pipeline thread to avoid retry timeouts
+                    // TODO: Cancellation Token Maybe?
+                    // TODO: Maybe pass
+                    childVariables = _psesHost.InvokeDelegate<VariableDetailsBase[]>
+                    (
+                        representation: "GetChildren",
+                        ExecutionOptions.Default,
+                        (_) =>
+                        {
+                            var existingRunspace = Runspace.DefaultRunspace;
```

Comment:
I'm just curious why you need to save, change, and then reset the runspace?

Reply:
@SeeminglyScience can comment but I think it's just a safety/idempotency thing, in all likelihood it's the same runspace so this basically doesn't do anything.

Reply:
Hm, if that's something we need to do perhaps we should do it in a wrapper? Like, where/when else might we have to do something like that?

And if we're not sure that we _have_ to do it, I'd rather leave a TODO for now so if we encounter a bug we fix it later.

Reply:
Yeah that was a troubleshooting step that didn't do anything right? If so that can be removed for sure.